pyalgostrategypool/strategy_options_short_iron_butterfly_with_reentry/_strategy.py

```diff
@@ -0,0 +1,181 @@
+
+        for param in (self.no_of_otm_strikes_buy_ce_leg, self.no_of_otm_strikes_sell_ce_leg, self.no_of_otm_strikes_buy_pe_leg, self.no_of_otm_strikes_sell_pe_leg):
+            check_argument(param, 'extern_function', is_positive_int, 'Number of strikes should be an integer > 0')
+
+        for param in [self.strategy_parameters.get("ALLOW_REENTRY_UPPER", 0), self.strategy_parameters.get("ALLOW_REENTRY_LOWER", 0)]:
+            check_argument(param, 'extern_function', lambda x: isinstance(x, int) and x in [0, 1], f'ALLOW_REENTRY parameters should be 0 (False) or 1 (True)')
+
+    def initialize(self):
+        super().initialize()
+
+        # Reset main orders for calls and puts
+        self.child_instrument_main_orders_ce, self.child_instrument_main_orders_pe = {}, {}
+        self.flag_lower_breach_possible_reentry = self.flag_upper_breach_possible_reentry = False
+        self.base_price_post_breach = None
+
+    def strategy_select_instruments_for_entry(self, candle, instruments_bucket):
+        selected_instruments, meta, executed_tradingsymbol_suffix = [], [], set()
+
+        for base_instrument in instruments_bucket:
+            self.logger.info(
+                f"Checking entry/re-entry conditions for base instrument: {base_instrument} | "
+                f"Determining OTM option instruments and verifying if CE/PE orders are already placed."
+            )
+            # Define a list of tuples for managing legs, their types, and relevant orders
+            leg_wise_list = [
+                ("ce_buy_leg", 'CE', self.no_of_otm_strikes_buy_ce_leg, 'BUY', self.child_instrument_main_orders_ce[base_instrument].get("ce_buy_leg") if self.child_instrument_main_orders_ce.get(base_instrument) else None),
+                ("ce_sell_leg", 'CE', self.no_of_otm_strikes_sell_ce_leg, 'SELL', self.child_instrument_main_orders_ce[base_instrument].get("ce_sell_leg") if self.child_instrument_main_orders_ce.get(base_instrument) else None),
+                ("pe_buy_leg", 'PE', self.no_of_otm_strikes_buy_pe_leg, 'BUY', self.child_instrument_main_orders_pe[base_instrument].get("pe_buy_leg") if self.child_instrument_main_orders_pe.get(base_instrument) else None),
+                ("pe_sell_leg", 'PE', self.no_of_otm_strikes_sell_pe_leg, 'SELL', self.child_instrument_main_orders_pe[base_instrument].get("pe_sell_leg") if self.child_instrument_main_orders_pe.get(base_instrument) else None)
+            ]
+
+            current_underlying_price = self.broker.get_ltp(base_instrument)
+
+            # Calculate re-entry conditions based on the latest price
+            flag_upper_re_entry = (self.flag_upper_breach_possible_reentry and (self.base_price_post_breach - current_underlying_price) >= self.price_breach_upper_offset)
+            flag_lower_re_entry = (self.flag_lower_breach_possible_reentry and (current_underlying_price - self.base_price_post_breach) >= self.price_breach_lower_offset)
```

Comment:
rearrange the equation such that the lhs is the same as the above one (mathematically, just multiply by -1 and rearrange)

Reply:
Done.